src/main/java/org/spongepowered/api/service/context/ContextCalculator.java

```diff
@@ -22,36 +22,37 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.service.permission.context;
-
-import org.spongepowered.api.service.permission.Subject;
+package org.spongepowered.api.service.context;
 
 import java.util.Set;
 
 /**
- * Calculate the availability of contexts. These methods may be invoked
- * frequently, and therefore should be fast.
+ * Calculate the availability of {@link Context}s for a {@link Contextual}. 
+ * These methods may be invoked frequently, and therefore should be fast.
+ * 
  */
 public interface ContextCalculator {
```

Comment:
This request has no correlation to economy. what you're asking for is generification of the current implementation regardless of the changes in this PR. I suggest you open a new issue for this.